### PR TITLE
GDB-9971 implements import resource filters

### PR DESCRIPTION
**Pull request by boyan-tonchev**
What
- Implements the 'File', 'File Folder', or 'Folder' filter options;
- Implements filtering by import rdf resource name.

Why
This will provide users with the opportunity to filter displayed import RDF data resources.

How
Added filtering for displayed resources based on user selection.

### CODE REVIEW

File: src/js/angular/utils/devounce.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Debounce {
```
Comment (svilenvelikov) [inline src/js/angular/utils/devounce.js:1]: fix the filename

File: src/js/angular/import/directives/import-resource-tree.directive.js [hunk 2]
```diff
@@ -22,30 +30,93 @@ function importResourceTreeDirective() {
...
+                resource.setSelection(resource.selected);
+                selectedResources = $scope.resources.getAllSelected();
+            });
+
+            $scope.changeTypeFilter = (newType) => {
+                $scope.filterByType = newType;
+                updateListedImportResources();
+            };
+
+            $scope.handleInputChange = function (filterByFileName) {
```
Comment (svilenvelikov) [inline src/js/angular/import/directives/import-resource-tree.directive.js:55]: Change to arrow function

Comment (svilenvelikov) [inline src/js/angular/import/directives/import-resource-tree.directive.js:55]: What input change? This should be `filterResources` I guess

File: src/js/angular/import/directives/import-resource-tree.directive.js [hunk 3]
```diff
@@ -22,30 +30,93 @@ function importResourceTreeDirective() {
...
 
             // =========================
             // Private variables
             // =========================
             let selectedResources = [];
 
+            // =========================
+            // Public functions
+            // =========================
+            $scope.selectionChanged = (($event, resource) => {
```
Comment (svilenvelikov) [inline src/js/angular/import/directives/import-resource-tree.directive.js:45]: If you don't need the event, then remove the parameter

File: src/js/angular/import/directives/import-resource-tree.directive.js [hunk 4]
```diff
@@ -22,30 +30,93 @@ function importResourceTreeDirective() {
...
+                resource.setSelection(resource.selected);
+                selectedResources = $scope.resources.getAllSelected();
+            });
+
+            $scope.changeTypeFilter = (newType) => {
```
Comment (svilenvelikov) [inline src/js/angular/import/directives/import-resource-tree.directive.js:50]: These three event handlers have very inconsistent names: selectionChanged, changeTypeFilter, handleInputChange and only the first one sounds like an event handler. Please, choose one convention and stick with it.

File: src/js/angular/models/import/import-resource-tree-element.js [hunk 5]
```diff
@@ -137,4 +137,54 @@ export class ImportResourceTreeElement {
...
     }
+
+    hasTextInResourcesName(searchText) {
+        if (this.importResource.name.includes(searchText)) {
+            return true;
+        }
+        if (this.files.some((file) => file.importResource.name.includes(searchText))) {
+            return true;
+        }
+        for (const directory of this.directories) {
```
Comment (svilenvelikov) [inline src/js/angular/models/import/import-resource-tree-element.js:148]: What's wrong with the `return this.directories.some((directory) => directory.hasTextInResourcesName(searchText))`? And same for the next two methods

File: src/js/angular/utils/devounce.js (hunk 1 above)
Comment (svilenvelikov) [reply src/js/angular/utils/devounce.js:1]: Where is this used?

Comment (boyan-tonchev) [reply src/js/angular/utils/devounce.js]: Not used, removed.

File: src/js/angular/import/directives/import-resource-tree.directive.js (hunk 2 above)
Comment (boyan-tonchev) [reply src/js/angular/import/directives/import-resource-tree.directive.js]: Done

Comment (boyan-tonchev) [reply src/js/angular/import/directives/import-resource-tree.directive.js]: Renamed.

File: src/js/angular/import/directives/import-resource-tree.directive.js (hunk 3 above)
Comment (boyan-tonchev) [reply src/js/angular/import/directives/import-resource-tree.directive.js]: Removed the event

**Review by svilenvelikov — APPROVED** [review]

**Pull request merged**